livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py

```diff
@@ -1023,6 +1031,38 @@ def _recover_from_text_response(self, item_id: str | None = None) -> None:
         self.conversation.item.create(self._create_empty_user_audio_message(1.0))
         self.response.create(on_duplicate="keep_both")
 
+    def _truncate_conversation_item(
+        self, item_id: str, content_index: int, audio_end_ms: int
+    ) -> None:
+        self.conversation.item.truncate(
+            item_id=item_id,
+            content_index=content_index,
+            audio_end_ms=audio_end_ms,
+        )
+
+    def _handle_response_content_done(self, message):
+        if message.content_type == "text":
+            if self._text_response_retries >= self._max_text_response_retries:
+                raise RuntimeError(
+                    f"The OpenAI Realtime API returned a text response "
+                    f"after {self._max_text_response_retries} retries. "
+                    f"Please try to reduce the number of text system or "
+                    f"assistant messages in the chat context."
+                )
+
+            self._text_response_retries += 1
+            logger.warning(
+                "The Realtime API returned a text response instead of audio.",
+                extra={
+                    "item_id": message.item_id,
+                    "text": message.text,
+                    "retries": self._text_response_retries,
+                },
+            )
+            self._recover_from_text_response(message.item_id)
+        else:
+            self._text_response_retries = 0
+
```

Comment:
I'm not sure about this, should we create a separate PR to discuss about it?

Reply:
Should I kepp `_truncate_conversation_item` it's similar to _push_audio